lmcache/cache_engine.py

```diff
@@ -294,29 +294,30 @@ def retrieve(
         self,
         tokens: torch.Tensor,
         mask: Optional[torch.Tensor] = None,
-    ) -> Tuple[KVCache, torch.Tensor]:
+        return_tuple: bool = True,
+    ) -> Tuple[Union[KVCache, torch.Tensor], torch.Tensor]:
         """
         Retrieve the KV cache of the tokens from the cache engine. The 
         retrieved KV cache should be a prefix of the input tokens.
 
         Input:
             tokens: the input tokens, with shape [seq_len]
 
-            format: either 'huggingface' or 'vllm'
-                For huggingface, it should have the shape of 
-                [num_heads, num_tokens, head_size]
-                
-                For vllm, it should have the shape of 
-                [num_tokens, num_heads, head_size]
-                
             mask: a boolean mask of tokens indicating which tokens'
             KV Cache should be retrieved. Currently, only support
             suffix mask.
 
-        Output:
+            return_tuple: whether to return the kv cache as a tuple or a 
+            single tensor
+
+        Output: 
             kv_tensors: the kv cache of the tokens, in the format of nested 
-                        tuples. Will be an empty tuple if no kv cache is 
-                        retrieved.
+            tuples ora single tensor with shape [num_layers, 2, hidden_dim, 
```

Comment:
Typo "ora". Also, maybe eventually we want to return them as a single big tensor?

Reply:
> Typo "ora".

Fixed the typo

> Also, maybe eventually we want to return a single big tensor for both cases?

Yeah, maybe we should discuss this. I think the goal is to eliminate the overhead of permutating the tensors.